### PR TITLE
server: change default loglevel to 'info', change some message levels to tune the noise

### PR DESCRIPTION
**Pull request by jmarantz**
*title*:
 > server: change default loglevel to 'info', change some message levels to tune the noise

*Description*:
> Changes warning messages issued on startup to Info messages.  The messages were only categorized as warnings so they would be emitted by default.  This PR changes the default loglevel to 'info', and adjusted several info messages down to debug, because we want them to be emitted during server operation.  This feature changes the default server behavior.

Fixes https://github.com/envoyproxy/envoy/issues/1854

*Risk Level*: Low
> The biggest risk is that some Info messages may remain that need to be demoted.  However, the Issue has a description of the analysis that was done.  A follow-up for remaining issues will be straightforward.  The biggest risk is that new PRs may introduce new Info messages that should be debug, so the Envoy community needs to be informed of this change.

*Testing*:
> All unit tests and integration tests were run.  Following integration tests, the logs were examined for excessive info messages.  Specifically:

  bazel test //test/integration:all --test_arg="-l info"
  bazel test //test/...

After the integration tests, analyzed the logs to determine whether there was too much noise:

  find . -name '*.log' -exec fgrep \[info\] {} \; >/tmp/info.log
  cut -f4-10 -d\] /tmp/info.log|cut -d\  -f1-2 | grep -v  test/ | sort | uniq -c | sort -n -r

This pipe shows me the sources of messages that come out most frequently in the integration tests. Full results below. I think these seem reasonable but happy to trim further. The remaining high frequency messages all seem to be per server startup or config, not per request.  Details are in a comment in https://github.com/envoyproxy/envoy/issues/1854

*Release Notes*:
 > Default loglevel changed to 'info'

Signed-off-by: Joshua Marantz <jmarantz@google.com>

### CODE REVIEW

**Comment by mattklein123** [comment]
Thank you @jmarantz! Will look tomorrow. FYI you will need to fix DCO. See the contributing guide. Apologize in advance for the non-optimal workflow.

**Comment by jmarantz** [comment]
Thanks Matt -- I'm not sure if I did that right.  I'll sort it out tomorrow as well.

**Comment by jmarantz** [comment]
I demoted the cluster-update messages to 'debug' (as shown in the most recent commit).  However I'm still having some git knowledge limitations and multiple-username issues with DCO.  I'm mulling over a full restart on this PR to sort that out.

Also the Mac CI failed (timeout) for reasons that seem unlikely to be related to this PR and it wasn't obvious how to force a retry.   I guess one is happening now due to the incremental commit.

**Comment by jmarantz** [comment]
I think I managed to sort out the DCO issues via https://stackoverflow.com/questions/1657017/how-to-squash-all-git-commits-into-one

Editing the PR description to remove markup.

**Comment by mattklein123** [comment]
@jmarantz cool LGTM. I would like to run this locally so will do so in the next couple of days and report back. Thank you this is great.

File: source/server/server.cc [hunk 1]
```diff
@@ -300,19 +300,19 @@ RunHelper::RunHelper(Event::Dispatcher& dispatcher, Upstream::ClusterManager& cm
 
   // Setup signals.
   sigterm_ = dispatcher.listenForSignal(SIGTERM, [this, &hot_restart, &dispatcher]() {
-    ENVOY_LOG(warn, "caught SIGTERM");
```
Comment (htuch) [inline source/server/server.cc]: I think these warning should show up at warn level, as they are exceptional events.

File: source/server/server.cc [hunk 2]
```diff
@@ -366,24 +366,24 @@ void InstanceImpl::run() {
...
 }
 
 void InstanceImpl::shutdownAdmin() {
-  ENVOY_LOG(warn, "shutting down admin due to child startup");
+  ENVOY_LOG(info, "shutting down admin due to child startup");
   stat_flush_timer_.reset();
   handler_->stopListeners();
   admin_->mutable_socket().close();
 
-  ENVOY_LOG(warn, "terminating parent process");
```
Comment (htuch) [inline source/server/server.cc]: Same for the log messages on the way out of the process.

File: source/server/server.cc (hunk 1 above)
Comment (jmarantz) [reply source/server/server.cc]: done

File: source/server/server.cc (hunk 2 above)
Comment (jmarantz) [reply source/server/server.cc]: done

**Pull request merged**